src/widlprocess.ts

```diff
@@ -0,0 +1,240 @@
+            ...convertIdlType(c.idlType),
+            param: c.arguments.map(convertArgument)
+        }]
+    }
+}
+
+function convertArgument(arg: webidl2.Argument): Browser.Param {
+    return {
+        name: arg.name,
+        ...convertIdlType(arg.idlType),
+        optional: arg.optional ? 1 : undefined,
+        variadic: arg.variadic ? 1 : undefined,
+    }
+}
+
+function convertAttribute(attribute: webidl2.AttributeMemberType): Browser.Property {
+    return {
+        name: attribute.name,
+        ...convertIdlType(attribute.idlType),
+        "read-only": attribute.readonly ? 1 : undefined
+    }
+}
+
+function convertConstantMember(constant: webidl2.ConstantMemberType): Browser.Constant {
+    return {
+        name: constant.name,
+        type: constant.idlType.idlType as string,
+        value: convertConstantValue(constant.value)
+    };
+}
+
+function convertConstantValue(value: webidl2.ValueDescription): string {
+    switch (value.type) {
+        case "string":
+            return `"${value.value}"`;
+        case "boolean":
+        case "number":
+        case "sequence":
+            return `${value.value}`;
+        case "null":
+        case "NaN":
+            return value.type;
+        case "Infinity":
+            return (value.negative ? '-' : '') + value.type;
+        default:
+            throw new Error("Not implemented");
+    }
+}
+
+function convertDictionary(dictionary: webidl2.DictionaryType) {
+    const result: Browser.Dictionary = {
+        name: dictionary.name,
+        extends: dictionary.inheritance || "Object",
+        members: { member: {} }
+    }
+    for (const member of dictionary.members) {
+        result.members.member[member.name] = convertDictionaryMember(member);
+    }
+    return result;
+}
+
+function convertDictionaryMember(member: webidl2.DictionaryMemberType): Browser.Member {
+    return {
+        name: member.name,
+        default: member.default ? convertConstantValue(member.default) : undefined,
+        ...convertIdlType(member.idlType)
+    }
+}
+
+function convertTypedef(typedef: webidl2.TypedefType): Browser.TypeDef {
+    return {
+        "new-type": typedef.name,
+        ...convertIdlType(typedef.idlType)
+    }
+}
+
+function convertIdlType(i: webidl2.IDLTypeDescription): Browser.Typed {
+    if (typeof i.idlType === "string") {
+        return {
+            type: i.idlType,
+            nullable: i.nullable ? 1 : undefined
+        };
+    }
+    if (i.generic) {
+        return {
+            type: i.generic,
+            subtype: convertIdlType(i.idlType as webidl2.IDLTypeDescription),
+            nullable: i.nullable ? 1 : undefined
+        };
+    }
+    if (i.union) {
+        return {
+            type: (i.idlType as webidl2.IDLTypeDescription[]).map(convertIdlType),
+            nullable: i.nullable ? 1 : undefined
+        };
+    }
+    throw new Error("Unsupported IDL type structure");
+}
+
+function createEmptyBrowserWebidl(): Browser.WebIdl {
```

Comment:
we have this function already in index.ts, i would move it helpers and reuse it.